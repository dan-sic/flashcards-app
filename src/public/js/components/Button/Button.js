import Button from "@material-ui/core/Button";
import { withStyles } from "@material-ui/core/styles";

export default withStyles({
  root: {
    "& .MuiButton-label": {
      fontSize: "1.5rem"
    }
  }
})(Button);
