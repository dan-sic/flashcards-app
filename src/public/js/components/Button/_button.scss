.button {
  background-color: $color-primary-dark;
  color: white;
  font-size: 1.6rem;
  font-family: inherit;
  padding: 1rem 3rem;
  border-radius: 0.3rem;
  border: none;
  margin: 0 auto;
  display: block;
}
