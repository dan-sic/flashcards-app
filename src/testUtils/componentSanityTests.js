import React from "react";
import { shallow } from "enzyme";

export const componentSanityTests = (Component, label) => {
  describe(`Sanity tests for ${label}`, () => {
    it("is defined", () => {
      expect(Component).toBeDefined();
    });

    it("renders element", () => {
      const wrapper = shallow(Component);
      expect(wrapper.isEmptyRender()).toBe(false);
    });

    it("matches the snapshot", () => {
      const wrapper = shallow(Component);
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
};
