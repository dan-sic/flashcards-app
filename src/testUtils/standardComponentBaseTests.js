import React from "react";
import { shallow } from "enzyme";

export const standardComponentBaseTests = (Component, label) => {
  describe(`Standard base test for ${label}`, () => {
    it("passes props to element", () => {
      const props = {
        onClick: jest.fn(),
        propValue: "button"
      };

      const wrapper = shallow(<Button {...props} />);
      wrapper.simulate("click");
      expect(props.onClick).toHaveBeenCalled();
      expect(
        wrapper
          .find("button")
          .first()
          .props()["type"]
      ).toEqual("button");
    });
  });
};
