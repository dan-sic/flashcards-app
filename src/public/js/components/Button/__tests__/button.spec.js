import React from "react";
import Button from "../Button";
import { shallow } from "enzyme";
import { componentSanityTests } from "../../../../../testUtils/componentSanityTests";

describe("Button", () => {
  componentSanityTests(<Button />, "Button");

  it("renders text passed as children props", () => {
    const text = "Hello";
    const wrapper = shallow(<Button>{text}</Button>);
    expect(wrapper.first().text()).toEqual(text);
  });

  it("passes props to button element", () => {
    const props = {
      onClick: jest.fn(),
      type: "button"
    };

    const wrapper = shallow(<Button {...props} />);
    wrapper.simulate("click");
    expect(props.onClick).toHaveBeenCalled();
    expect(
      wrapper
        .find("button")
        .first()
        .props()["type"]
    ).toEqual("button");
  });
});
