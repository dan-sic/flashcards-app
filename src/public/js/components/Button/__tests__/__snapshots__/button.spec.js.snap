// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Sanity tests for Button matches the snapshot 1`] = `"<button class=\\"button\\"></button>"`;
